frontend/src/app/upload.service.ts

```diff
@@ -0,0 +1,54 @@
+/* Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http'
+import { BehaviorSubject } from 'rxjs'
+
+@Injectable({
+  providedIn: 'root'
+})
+/**
+ * UploadService handles communication with the backend.
+ * When a response is received, it shares it to other components
+ * through a shared message system.
+ */
+export class UploadService {
+  serverUrl: string = "http://localhost:5000/upload"
```

Comment:
I assume this will change once we move this online? Might be useful to document this somewhere as something to change or make configurable at some point.

Reply:
Yes this will change when not running locally, so I'll comment it